internal/controller/sync.go

```diff
@@ -352,19 +357,127 @@ func (r *Route) buildNextCR(ctx context.Context, route *routev1.Route, revision
 			uriSans = append(uriSans, ur)
 		}
 	}
+	var emailAddresses []string
+	if metav1.HasAnnotation(route.ObjectMeta, cmapi.EmailsAnnotationKey) {
+		emailAddresses = strings.Split(route.Annotations[cmapi.EmailsAnnotationKey], ",")
+	}
+	var organizations []string
+	if metav1.HasAnnotation(route.ObjectMeta, cmapi.SubjectOrganizationsAnnotationKey) {
+		subjectOrganizations, err := cmutil.SplitWithEscapeCSV(route.Annotations[cmapi.SubjectOrganizationsAnnotationKey])
+		organizations = subjectOrganizations
+
+		if err != nil {
+			r.log.V(1).Error(err, "the organizations annotation is invalid",
+				"object", route.Namespace+"/"+route.Name, cmapi.SubjectOrganizationsAnnotationKey,
+				route.Annotations[cmapi.SubjectOrganizationsAnnotationKey])
+			r.eventRecorder.Event(route, corev1.EventTypeWarning, ReasonInvalidValue, "annotation "+cmapi.SubjectOrganizationsAnnotationKey+": "+route.Annotations[cmapi.SubjectOrganizationsAnnotationKey]+" value is malformed")
+			return nil, err
+		}
+	}
+	var organizationalUnits []string
+	if metav1.HasAnnotation(route.ObjectMeta, cmapi.SubjectOrganizationalUnitsAnnotationKey) {
+		subjectOrganizationalUnits, err := cmutil.SplitWithEscapeCSV(route.Annotations[cmapi.SubjectOrganizationalUnitsAnnotationKey])
+		organizationalUnits = subjectOrganizationalUnits
+
+		if err != nil {
+			r.log.V(1).Error(err, "the organizational units annotation is invalid",
+				"object", route.Namespace+"/"+route.Name, cmapi.SubjectOrganizationalUnitsAnnotationKey,
+				route.Annotations[cmapi.SubjectOrganizationalUnitsAnnotationKey])
+			r.eventRecorder.Event(route, corev1.EventTypeWarning, ReasonInvalidValue, "annotation "+cmapi.SubjectOrganizationalUnitsAnnotationKey+": "+route.Annotations[cmapi.SubjectOrganizationalUnitsAnnotationKey]+" value is malformed")
```

Comment:
**Suggestion:** Same as in https://github.com/cert-manager/openshift-routes/pull/29/files#r1326934090, let's add more context to the `err` and return it as-as. The caller of `buildNextCR` will deal with firing an event and logging the error.